### PR TITLE
TST: fix a couple broken test cases

### PR DESCRIPTION
**Pull request by neutrinoceros**
### Description
I discovered while working on #17179 that a couple test cases were silently broken since #16433 when run against numpy>=2.0

This is admittedly confusing (which explains why I did this mistake and why it passed review), so let me try to explain what's going on:
Because of missing parens, these assert statements are currently equivalent to
```python
assert (1 == 2 if condition else 1)
```

though the way the Python intepreter reads the expression `(1 == 2 if False else True)`, it will *first* evaluate the condition (`if condition`) and then, with `condition==False`, jump to the `else` branch as the result of the expression: `1`, which is "truthy", resulting in a successful assertion, never actually evaluating the (here, broken) comparison (`1 == 2`).
In short, the current form reads as
```python
assert ((1 == 2) if condition else (1))
```

The correct, intended form is
```python
assert 1 == (2 if condition else 1)
```

- [ ] By checking this box, the PR author has requested that maintainers do **NOT** use the "Squash and Merge" button. Maintainers should respect this when possible; however, the final decision is at the discretion of the maintainer that merges the PR.

### CODE REVIEW

**Review by pllim — APPROVED** [review]
Makes sense. Thanks!

p.s. If you milestone to backport branch release, then please also apply the backport label.
